src/commcare_cloud/ansible/roles/commcarehq/tasks/celery.yml

```diff
@@ -53,8 +53,55 @@
     - services
   register: celery_errand_boy_files
 
+- name: Add celery_bash_runner files
+  template:
+    src: "../templates/celery_bash_runner.sh.j2"
+    dest: "{{ service_home }}/{{ deploy_env }}_{{ item.file_name }}"
+  with_items:
+    - file_name: "celery_bash_runner.sh"
+    - file_name: "celery_bash_runner_optimized.sh"
+      python_options: "-O"
+  tags:
+    - services
+  register: celery_bash_runner_files
+
+- name: define celery_flower services
+  become: cchq
+  template:
+    src: "../templates/supervisor_celery_flower.conf.j2"
+    dest: "{{ service_home }}/{{ deploy_env }}_supervisor_celery_flower.conf"
+  when: app_processes_config.celery_processes.get(inventory_hostname).flower is defined
+  tags:
+    - services
+  register: celery_flower_files
+
+- name: define celery beat service
+  become: cchq
+  template:
+    src: "../templates/supervisor_celery_beat.conf.j2"
+    dest: "{{ service_home }}/{{ deploy_env }}_supervisor_celery_beat.conf"
+  when: app_processes_config.celery_processes.get(inventory_hostname).celery_periodic is defined
```

Comment:
Historically fabric required beat to be on the same server as the periodic queue. I don't think that's necessary and I think it would limit us to only having the periodic queue on one server. I think we should be able to have multiple of those as long as there's only one beat process?

Reply:
Makes sense to me. As far as I understand all `beat` does is send tasks into the queue so it shouldn't matter where the queue worker is.